Feature/store search

oneul0 opened the pull request:
## 변경 타입
- [x] 신규 기능 추가/수정
- [ ] 버그 수정
- [x] 리팩토링
- [x] 설정
- [x] 비기능 (주석 등 기능에 영향을 주지 않음)

## 변경 내용
- **as-is**
  - 가게 검색 미추가 버전

- **to-be**
  - 등록된 모든 가게 기능 검색 api 구현
  - 요청한 위치에서 반경 3km 이내의 가게 검색 기능 추가(카테고리 미지정)
  - build.gradle 추가 내용
      - 맥 환경에서 서버 실행 시 나는 에러 잡기 위해 netty 의존성 추가
      - 쿼리 dsl 을 이용해서 postgis 기능 이용하기 위한 hibernate-spatial 의존성 추가
      - 쿼리 dsl 의존성 추가
  - postgis 설정 추가(docker-compose.yaml)
  - 응답 값에 거리가 포함된 dto, StoreNearbyStoreResponseDto 추가
  - 쿼리 dsl을 이용하기 위한 커스텀 Repository StoreRepositoryCustom와 구현체 StoreRepositoryImpl 추가
  - Store 서비스에서 유저 권한 체크하는 부분 주석 삭제(추후 userdetail 파라미터도 삭제 예정)

## 코멘트
- (추가적인 설명이나 코멘트가 필요한 경우 여기에 작성)

codejomo99 committed:
feat: 로그인 기능 구현

codejomo99 committed:
feat: JwtFilter 생성 후 config 수정

codejomo99 committed:
feat: JwtFilter 생성 후 config 수정

codejomo99 committed:
feat: 로그인 검증 생성
- 1. 클라이언트가 signIn api 요청
- 2. 유저 검증 후 jwt 생성해서 header,body 보내줌

codejomo99 committed:
feat: JwtFilter 수정
- 1. 토큰 생성 및 검증은 api 에서 처리
- 2. 필터에서는 인증 인가만 하도록 수정

codejomo99 committed:
feat: JwtFilter 수정
- 1. 토큰 생성 및 검증은 api 에서 처리
- 2. 필터에서는 인증 인가만 하도록 수정

oneul0 committed:
feat: 가게 등록 api에서 가게 상호명 중복 체크 기능 추가

oneul0 committed:
feat: 가게 등록 api에서 가게 상호명 중복 체크 기능 추가

oneul0 committed:
remove: region 디렉토리 삭제

oneul0 committed:
remove: region 디렉토리 삭제

Commit:
feat: updateOrderStateToCancel 기능구현

# Conflicts:
#	src/main/java/com/sparta/deliveryapp/order/controller/OrderController.java
#	src/main/java/com/sparta/deliveryapp/order/entity/Order.java

Commit:
feat: updateOrderStateToCancel 기능구현

# Conflicts:
#	src/main/java/com/sparta/deliveryapp/order/controller/OrderController.java
#	src/main/java/com/sparta/deliveryapp/order/entity/Order.java

Commit:
fix: updateOrderStateToCancel 트랜잭션 추가

Commit:
fix: order 필드 및 dto 수정

Commit:
fix: order 필드 및 dto 수정

Commit:
fix: order userId로 변경 및 그에 따른 dto 수정

Commit:
fix: order userId로 변경 및 그에 따른 dto 수정

Commit:
comment: updateOrderStateToCancel 서비스,컨트롤러 주석 변경

oneul0 committed:
feat: 가게 정보 등록 api 구현(유저 인증 기능 미구현)

oneul0 committed:
feat: 가게 정보 등록 api 구현(유저 인증 기능 미구현)

minji-git committed:
fix: init gradle-wrapper.properties

oneul0 committed:
feat: delete mapping 작성

# Conflicts:
#	src/main/java/com/sparta/deliveryapp/store/controller/StoreController.java

oneul0 committed:
feat: delete mapping 작성

codejomo99 committed:
feat: 회원유저 정보 Soft 삭제

codejomo99 committed:
feat: 회원유저 정보 Soft 삭제

Commit:
fix: 로그인 정보 인증객체 추가

Commit:
chore: 오류 메시지 수정

Commit:
comment: 주석 변경

Commit:
rename: OrderToCanceled dto 파일명 변경

Commit:
rename: OrderToCanceled dto 파일명 변경

Commit:
rename: dto 및 OrderStatusService 수정

Commit:
rename: dto 및 OrderStatusService 수정

oneul0 committed:
feat: 가게 등록 api 구현

카카오 로컬 api 연동 및 경위도 추출 메서드 구현
가게 엔티티 변수명과 json 변수명 매칭

oneul0 committed:
feat: 가게 등록 api 구현

카카오 로컬 api 연동 및 경위도 추출 메서드 구현
가게 엔티티 변수명과 json 변수명 매칭

oneul0 committed:
test: storeService 테스트 코드 추가

권한 체크 테스트
동일 상호명이 존재할 때 테스트
시간 변환 메서드 테스트

oneul0 committed:
test: storeService 테스트 코드 추가

권한 체크 테스트
동일 상호명이 존재할 때 테스트
시간 변환 메서드 테스트

oneul0 committed:
feat: 가게 엔티티에 베이스엔티티 적용

oneul0 committed:
rename: store 엔티티 이름 변경

storeEntity -> store

haaneul11 committed:
rename : Repository 파일명 수정

haaneul11 committed:
rename : Repository 파일명 수정

codejomo99 committed:
feat: User 조회 기능 구현

codejomo99 committed:
feat: User 조회 기능 구현

oneul0 committed:
feat: 이름으로 검색 api 추가, master 컨트롤러 분리

- store 엔티티에 Getter 추가
- 가게 삭제 api master controller로 이동

oneul0 committed:
feat: 이름으로 검색 api 추가, master 컨트롤러 분리

- store 엔티티에 Getter 추가
- 가게 삭제 api master controller로 이동

codejomo99 committed:
feat: User , AI 연관매핑 추가

codejomo99 committed:
feat: User , AI 연관매핑 추가

oneul0 committed:
chore: dev 브랜치와 병합, repository id uuid로 변경

codejomo99 committed:
chore: beanutils 의존성 추가

oneul0 committed:
feat: 가게 이름으로 모든 가게 정보 검색 기능 개발

oneul0 committed:
feat: 가게 이름으로 모든 가게 정보 검색 기능 개발

oneul0 committed:
feat: 가게 이름으로 모든 가게 정보 검색 기능 개발

oneul0 committed:
feat: 가게 id로 검색 기능 구현

oneul0 committed:
SCRUM-25 feat: 경위도로 가게 찾기 기능 추가

- postgis 설정을 위한 docker-compose.yaml 설정 추가
- storecontroller에 경위도로 가게 찾기 api 추가
- Store 엔티티에 경위도 검색을 위한 Point 임시 컬럼 추가
- build.gradle에 netty, queryDSL, hibernate-spatial 의존성 추가

oneul0 committed:
SCRUM-25 feat: 경위도로 가게 찾기 기능 추가

- postgis 설정을 위한 docker-compose.yaml 설정 추가
- storecontroller에 경위도로 가게 찾기 api 추가
- Store 엔티티에 경위도 검색을 위한 Point 임시 컬럼 추가
- build.gradle에 netty, queryDSL, hibernate-spatial 의존성 추가

codejomo99 committed:
refactor: 불 필요한 import

codejomo99 committed:
refactor: 불 필요한 import

oneul0 committed:
feat: 카테고리 미지정 시 근처 가게 목록 조회 기능 구현

oneul0 committed:
feat: 카테고리 미지정 시 근처 가게 목록 조회 기능 구현

oneul0 committed:
feat: 카테고리 미지정 시 근처 가게 목록 조회 기능 쿼리 dsl로 변경

oneul0 committed:
chore: 주석 정리

oneul0 committed:
chore: 주석 정리

codejomo99 commented on src/main/java/com/sparta/deliveryapp/store/controller/StoreMasterController.java:
검증 로직을 잘 분리하고, 컨트롤러 단에서 비즈니스 로직을 최소화해서 깔끔하게 작성한 점이 인상 깊습니다. 유지보수성과 가독성이 높아졌네요! 👍

codejomo99 commented on src/main/java/com/sparta/deliveryapp/store/repository/StoreRepositoryImpl.java:
QueryDSL 과 PostGIS 를 사용한 위치 기반 코드를 잘 작성한거 같습니다 !!

Sodychoe commented on build.gradle:
```diff
@@ -52,8 +52,41 @@ dependencies {
+    annotationProcessor "jakarta.annotation:jakarta.annotation-api" // java.lang.NoClassDefFoundError (javax.annotation.Generated) 대응 코드
+    annotationProcessor "jakarta.persistence:jakarta.persistence-api" // java.lang.NoClassDefFoundError (javax.annotation.Entity) 대응 코드
+
+    //mac 환경을 위한 netty
+    runtimeOnly 'io.netty:netty-resolver-dns-native-macos:4.1.104.Final:osx-aarch_64'
```
특정 플랫폼에 종속된 의존성이니 깃허브에는 올리지 않는 건 어떨까요??
git stash 등을 이용하면 로컬에서만 사용할 수 있을 것 같아요

oneul0 replied:
넵! 방법 찾아서 수정하고 다시 커밋 올리겠습니다!

Sodychoe commented on src/main/java/com/sparta/deliveryapp/store/service/StoreService.java:
```diff
@@ -86,6 +80,99 @@ public Store deleteStore(String storeId, UserDetailsImpl userDetails) {
+      double latitude, UserDetailsImpl userDetails) {
+    final int RANGE = 3000;
+
+    if (getDecimalPlaces(longitude) < 2 || getDecimalPlaces(latitude) < 2) {
+      throw new IllegalArgumentException("주어진 좌표의 자릿수가 너무 작습니다.");
+    }
```
무엇을 위한 예외처리인지 궁금합니다!

oneul0 replied:
postgis의 ST_Distance 메서드를 이용하여 두 좌표 간의 곡선 거리를 측정할 때, 비교 좌표(store 테이블에 저장된 소숫점 아래 12자리로 저장된 실수)와 기준 좌표(사용자의 좌표)의 오차가 많이 나면 조회가 정상적으로 되지 않는 문제가 발생하여 작성한 예외 처리 코드입니다.☺️
테스트를 통해 최대 10^-10 까지는 오차가 허용되는 것을 확인하여 소숫점 아래 2자리 미만인 좌표가 입력되게 되면 오류를 던지게 만들었는데 하드코딩한 느낌이라 나중에 리팩토링이 좀 필요할 것 같습니다!

minji-git commented on src/main/java/com/sparta/deliveryapp/store/controller/StoreMasterController.java:
```diff
@@ -0,0 +1,46 @@
+
+  //이름으로 가게 목록 조회
+  @GetMapping("/")
+  public ResponseEntity<List<StoreResponseDto>> findStoresByName(
```
조회 기능은 List로 넘겨주고, 서치 기능만 페이지로 나타내면 될까요?

minji-git commented on src/main/java/com/sparta/deliveryapp/store/dto/StoreResponseDto.java:
```diff
@@ -18,8 +18,8 @@
-//응답 값에 거리가 포함되지 않은
-public class StoreResponseDto extends BaseEntity {
+//응답 값에 거리가 포함되지 않은 가게 정보
+public class StoreResponseDto {
```
BaseEntity 사용하지 않는 건가요??

minji-git reviewed (commented):
확실히 queryDSL 복잡하고 어렵네요..

oneul0 committed:
feat: 카테고리 없이 근처 가게 조회 페이지네이션 기능 추가

- storecontroller 에 PageableDefault 어노테이션 추가
- storeService 반환타입 Page 객체로 변환
- store repository에 pageable 파라미터 추가
- storeRepository 구현체 변경
    - 쿼리 결과 반환 자료구조 변경(List<Tuple> -> QueryResults<Tuple>)
    - 반환 타입 변경(List<responseDto> -> Page<responseDto>)
- storeController getNearbyStoresWithoutCategory api 경위도 requestparams로 변경

oneul0 committed:
feat: 카테고리 없이 근처 가게 조회 페이지네이션 기능 추가

- storecontroller 에 PageableDefault 어노테이션 추가
- storeService 반환타입 Page 객체로 변환
- store repository에 pageable 파라미터 추가
- storeRepository 구현체 변경
    - 쿼리 결과 반환 자료구조 변경(List<Tuple> -> QueryResults<Tuple>)
    - 반환 타입 변경(List<responseDto> -> Page<responseDto>)
- storeController getNearbyStoresWithoutCategory api 경위도 requestparams로 변경

oneul0 committed:
feat: 카테고리 없이 근처 가게 조회 페이지네이션 기능 추가

- storecontroller 에 PageableDefault 어노테이션 추가
- storeService 반환타입 Page 객체로 변환
- store repository에 pageable 파라미터 추가
- storeRepository 구현체 변경
    - 쿼리 결과 반환 자료구조 변경(List<Tuple> -> QueryResults<Tuple>)
    - 반환 타입 변경(List<responseDto> -> Page<responseDto>)
- storeController getNearbyStoresWithoutCategory api 경위도 requestparams로 변경

oneul0 committed:
feat: 카테고리 없이 근처 가게 조회 페이지네이션 기능 추가

- storecontroller 에 PageableDefault 어노테이션 추가
- storeService 반환타입 Page 객체로 변환
- store repository에 pageable 파라미터 추가
- storeRepository 구현체 변경
    - 쿼리 결과 반환 자료구조 변경(List<Tuple> -> QueryResults<Tuple>)
    - 반환 타입 변경(List<responseDto> -> Page<responseDto>)
- storeController getNearbyStoresWithoutCategory api 경위도 requestparams로 변경

oneul0 committed:
feat: 카테고리 없이 근처 가게 조회 페이지 정렬 조건 거리 가까운 순으로 수정

oneul0 committed:
feat: 이름으로 가게 목록 조회 페이지네이션 적용

oneul0 committed:
feat: 이름으로 가게 목록 조회 페이지네이션 적용

oneul0 committed:
feat: 이름으로 가게 목록 조회 페이지네이션 적용

oneul0 committed:
feat: 이름으로 가게 목록 조회 페이지네이션 적용

oneul0 committed:
chore: build gradle beanutil 빠진 의존성 추가 및 store 주석 및 userdtails 정리

oneul0 committed:
chore: rebase 후 dev 브랜치 파일로 변경
